src/Microsoft.Extensions.DependencyInjection.Abstractions/EnumerableServiceDescriptor.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public class EnumerableServiceDescriptor: ServiceDescriptor
```

Comment:
:clap:

From https://github.com/aspnet/DependencyInjection/pull/416#issuecomment-233189947:

> BTW, why are these properties on same type instead of different types?

How do we get to these types from the collection? Just enumerate and switch on type? Since this is a breaking change, does this mean 2,0 is coming pretty soon after 1.1? :smile: